instrumentation/opentelemetry-instrumentation-logging/src/opentelemetry/instrumentation/logging/__init__.py

```diff
@@ -0,0 +1,211 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The OpenTelemetry ``logging`` integration automatically injects tracing context into log statements.
+
+The integration patches the standard library logging module to automatically inject tracing context into log record objects
+and optionally calls ``logging.basicConfig()`` to set a logging format with placeholders for span ID, trace ID and service name.
+
+The following keys are injected into the patched log record objects:
+
+- ``otelSpanID``
+- ``otelTraceID``
+- ``otelServiceName``
+
+The integration uses the following logging format by default:
+
+.. code-block::
+
+    %(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] [trace_id=%(otelTraceID)s span_id=%(otelSpanID)s service.name=%(otelServiceName)s] - %(message)s
+
+Enable trace context injection
+------------------------------
+
+The integration is opt-in and must be enabled explicitly by setting the environment variable ``OTEL_PYTHON_LOG_CORRELATION`` to ``true``.
+
+The integration always patches the logging module to inject tracing context into the log record objects. Setting ``OTEL_PYTHON_LOG_CORRELATION`` to ``true``
+calls ``logging.basicConfig()`` to set a logging format that actually makes use of the injected variables.
+
+
+Environment variables
+---------------------
+
+OTEL_PYTHON_LOG_CORRELATION
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This env var must be set to ``true`` in order to enable trace context injection into logs by calling ``logging.basicConfig()`` and
+setting a logging format that makes use of the injected tracing variables.
+
+The default value is ``false``.
+
+OTEL_PYTHON_LOG_FORMAT
+^^^^^^^^^^^^^^^^^^^^^^
+
+This env var can be used to instruct the instrumentation to use a custom logging format.
+
+The default value is:
+
+.. code-block::
+
+    %(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] [trace_id=%(otelTraceID)s span_id=%(otelSpanID)s service.name=%(otelServiceName)s] - %(message)s
```

Comment:
can this just link to the above default format? copy-paste can make it difficult to maintain code.

Reply:
Not sure how linking will work in docs. Do you mean link to variable definition directly to source/github or define this string only once in docs with an anchor and link to that from everywhere else? 

I tried a different solution. I'm assigning docstrings explicitly by setting `__doc__ = "....".format(DEFAULT_LOGGING_FORMAT)`. This way the docs will always be up to date and logging format will have a single source of truth for both code and docs. It feels a bit weird but technically it's not inferior in any way. I can't think of a downside but may be I didn't think hard enough.

What do you think about this approach? 

https://github.com/open-telemetry/opentelemetry-python-contrib/pull/345/files#diff-b3cc41cf4c57f13f0ac89e4d3edec4ce0156b65ed65e94fbd95660d1f763ae22R37
and
https://github.com/open-telemetry/opentelemetry-python-contrib/pull/345/files#diff-b3cc41cf4c57f13f0ac89e4d3edec4ce0156b65ed65e94fbd95660d1f763ae22R48